### PR TITLE
Add delete action to module manager

### LINKED ISSUE #30101: Modules page default button changed on 8.0
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

If module is not configurable, the default module action changed from "Disable" to "Disable on mobile" on 8.0.
There was an attempt to fix it here - https://github.com/PrestaShop/PrestaShop/pull/28332 (issue https://github.com/PrestaShop/PrestaShop/issues/28331), but it wasn't fixed properly.

**1.7.6.9**

**8.0 RC**
RC

### Expected behavior

Default action for non configurable, non upgradable module should be **Disable**.

### Steps to reproduce

1. Install 1.7.6 or 1.7.7 or 1.7.8 branch.
2. Install 8.0 RC.
3. Open Module manager and see that there is a different default module action.

### PrestaShop version(s) where the bug happened

8.0

### PHP version(s) where the bug happened

8.0

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | This PR adds an option to delete an uninstalled module. Before, you had to install it and uninstall again with "delete" checkbox enabled in the modal. Also working for bulk actions. Keeping a PM label, we talked about it on Slack with @MatShir once and he didn't even knew that this was not implemented. :-)
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Uninstall a module without deleting, refresh page. See that you can now delete the module from disk right away.
| Fixed ticket?     | 
| Related PRs       | 
| Sponsor company   | [TRENDO s.r.o.](https://www.trendo.cz)

### For QA, there are known issue that will be fixed in another PR. They are in current develop, not caused by this PR.
- https://github.com/PrestaShop/PrestaShop/issues/30952
- https://github.com/PrestaShop/PrestaShop/issues/30953

### How does it look
delete

### CODE REVIEW

**Comment by MatShir** [comment]
I'm watching you 
image

**Comment by MatShir** [comment]
If you could provide some screenshots, it would be perfect. It would help the QA testing it faster ;)

File: src/Core/Module/ModuleManagerInterface.php [hunk 1]
```diff
@@ -50,6 +50,8 @@ public function disableMobile(string $name): bool;
 
     public function reset(string $name, bool $keepData = false): bool;
 
+    public function delete(string $name): bool;
```
Comment (kpodemski) [inline src/Core/Module/ModuleManagerInterface.php]: BC Break? 🤯 Nitpick but...

Comment (Hlavtox) [reply src/Core/Module/ModuleManagerInterface.php]: Whaaaaat? It's a new function :D

Comment (FabienPapet) [reply src/Core/Module/ModuleManagerInterface.php]: Of course it's a bc break, but you can get rid of it 
see `\PrestaShop\PrestaShop\Core\Util\String\StringModifierInterface`

**Review by FabienPapet — CHANGES_REQUESTED** [review]

**Comment by Hlavtox** [comment]
@FabienPapet @kpodemski BC break resolved, can you check?

**Review by FabienPapet — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Review by AureRita — APPROVED** [review]
Hi @Hlavtox 

Thank you for your PR, I tested it and it seems to works as we can see here : 

[Untitled_ Jan 31, 2023 12_22 PM.webm](https://user-images.githubusercontent.com/100699445/215780008-952a5eb3-7364-45b7-a739-ecca0cf9e042.webm)

thank you

**Comment by Hlavtox** [comment]
Thanks @AureRita!

**Comment by kpodemski** [comment]
@Hlavtox as you can see on the video it could be improved, so that after the Delete action you delete the row in HTML :)

**Comment by MatShir** [comment]
We can create a bug, so it can be fixed during the beta. wdyt? @Hlavtox

**Comment by Hlavtox** [comment]
@MatShir @kpodemski We will need to check it with @atomiix. The hiding functionality worked before. It must be removed from the list only if the module is not offered by a distribution API.

**Pull request merged**